frontend/lib/resolution/resolution-queries.cpp

```diff
@@ -3575,6 +3575,37 @@ bool isTypeDefaultInitializable(Context* context, const Type* t) {
   return isTypeDefaultInitializableQuery(context, t);
 }
 
+static const QualifiedType&
+computeCustomInferTypeQuery(Context* context,
+                            const CompositeType* t,
+                            const Scope* inScope,
+                            const PoiScope* inPoiScope) {
```

Comment:
Good call here. Looking at things more closely, I realized that we'll want to codegen this method for arrays because of the runtime types. With that in mind, this makes much more sense as an AssociatedAction.